required/latex-lab/latex-lab-mathintent.dtx

```diff
@@ -0,0 +1,109 @@
+% \iffalse meta-comment
+%
+%% File: latex-lab-mathintent.dtx
+%
+% Copyright (C) 2025 The LaTeX Project
+%
+% It may be distributed and/or modified under the conditions of the
+% LaTeX Project Public License (LPPL), either version 1.3c of this
+% license or (at your option) any later version.  The latest version
+% of this license is in the file
+%
+%    https://www.latex-project.org/lppl.txt
+%
+%
+% The development version of the bundle can be found below
+%
+%    https://github.com/latex3/latex2e/required/latex-lab
+%
+% for those people who are interested or want to report an issue.
+%
+%<*driver>
+\documentclass{l3doc}
+\EnableCrossrefs
+\CodelineIndex
+\begin{document}
+  \DocInput{latex-lab-mathintent.dtx}
+\end{document}
+%</driver>
+%
+% \fi
+%
+%
+% \title{The \texttt{latex-lab-mathintent} code\thanks{}}
+% \author{\LaTeX{} Project}
+%
+% \maketitle
+%
+% \newcommand\fmi[1]{\begin{quote} TODO: \itshape #1\end{quote}}
+% \newcommand\NEW[1]{\marginpar{\mbox{}\hfill\fbox{New: #1}}}
+% \providecommand\class[1]{\texttt{#1.cls}}
+% \providecommand\pkg[1]{\texttt{#1}}
+% \providecommand\hook[1]{\texttt{#1}}
+%
+% \begin{abstract}
+% \end{abstract}
+%
+% \tableofcontents
+%
+% \section{Introduction}
+%
+% This file implements Experiments with the MathML5 \texttt{intent} feature.
+% Initially to give readings fro invisible times and function application.
+%
+% \MaybeStop{\setlength\IndexMin{200pt}  \PrintIndex  }
+%
+% \section{The Implementation}
+%
+%    \begin{macrocode}
+%<@@=math>
+%    \end{macrocode}
+%
+%    \begin{macrocode}
+%<*kernel>
+%    \end{macrocode}
+%
+% \subsection{File declaration}
+%    \begin{macrocode}
+\ProvidesFile{latex-lab-mathintent.ltx}
+        [2025-01-17 v0.1a intent experiments]
+%    \end{macrocode}
+%
+% \subsection{Invisible ops}
+%    \begin{macrocode}
+\ExplSyntaxOn
+%    \end{macrocode}
+%
+%    \begin{macrocode}
+\protected\def\invisibletimes{
+\if_cs_exist:N\luamml_annotate:en
+  \luamml_annotate:en {
+   core = {[0] = 'mo', intent="times","^^^^2062"},
+}{
+ \latelua{}
+}
+\fi:
```

Comment:
Shouldn't these be `\cs_new_protected`? Also, can't this be done late enough that they are expanded to a single definition rather than re-checking every time they are used?

Reply:
in the end, probably yes (but in the end they wouldn't be in latex-lab either)  the timing is a bit tricky as luamml is (normally) only implicitly loaded if suitable tagging options are set up and unicode-math has  been loaded by begin document, so you'd have to delay the definition until then, which is possible of course, although the main concern here was finding something that works, ie passing luamml information without affecting the typeset result. I started with `\protected\def` as  I was going to suggest it as an addition for amsmath so it was always available (as a no-op   (but then I needeed to test for the l3 luamml any anyway so it's a bit hybrid.